lib/modules/datasource/packagist/schema.ts

```diff
@@ -0,0 +1,97 @@
+import { z } from 'zod';
+import { api as versioning } from '../../versioning/composer';
+import type { Release, ReleaseResult } from '../types';
+
+export const ComposerRelease = z.object({
+  version: z
+    .string()
+    .refine((v) => versioning.isSingleVersion(v), 'Invalid version'),
+  homepage: z.string().nullable().catch(null),
+  source: z
+    .object({
+      url: z.string(),
+    })
+    .transform((x) => x.url)
+    .nullable()
+    .catch(null),
+  time: z.string().nullable().catch(null),
+});
+
+export const ComposerReleaseArray = z
+  .array(ComposerRelease.nullable().catch(null))
+  .transform((xs) =>
+    xs.filter((x): x is z.infer<typeof ComposerRelease> => x !== null)
+  );
+export type ComposerReleaseArray = z.infer<typeof ComposerReleaseArray>;
```

Comment:
are you sure we want the const and type having same name? 🤔

Reply:
What downsides you can see? As far as I can see, they reside in different namespaces that are quite orthogonal. What can go wrong?

Reply:
won't typescript merge both types, which is not the behavior we want? 🤔

Reply:
I was okay using same name, my dev experience was that it knows what does it mean in every context

Reply:
ok, still looks strange to me. 🤔